src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/StyleHelper.cs

```diff
@@ -5466,9 +5472,13 @@ internal static DependencyObject GetChild(List<DependencyObject> styledChildren,
         //
         internal static void RegisterAlternateExpressionStorage()
         {
-            DependencyObject.RegisterForAlternativeExpressionStorage(
+            if(_getExpression == null)
```

Comment:
After we register in static constructor, the `_getExpression` never be null when the other code call this method. And I think the `RegisterAlternateExpressionStorage` can be private and only call from static constructor.

Reply:
@lindexi - Actually this method gets called from `Style` class's static ctor as well. That's when we'll have a case where `_getExpression` has already setup via `StyleHelper`'s static ctor. Hence, this check prevents dual registration of the delegate.